design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,192 @@
+
+[Wireframe 4](https://share.balsamiq.com/c/hNWzHEM9ksGGp5ZcPixLrA.png)
+
+### User adds multiple select column steps
+
+- Users might want to add more than one `Select Columns` step.
+- Take a scenario where formula columns were created for the purpose of using them in other computed columns. The user might want to remove them from the result table. Using an additional select column that follows those `Add Formula Column` steps a user can define the columns to be included.
+
+## Workflow: Add Formula Column
+
+In the `Add Formula Column` step, users apply formulas to generate new columns.
+
+### User adds a formula column
+
+#### From the Worflow Sidebar
+
+- To add a formula column, the user can click on the `Add` button located on top of the `Workflow Sidebar` and select the `Generate Column` option.
+
+[Wireframe 5](https://share.balsamiq.com/c/qYNJBmBwdaTDhn7Hdhf9kP.png)
+
+#### From the result table
+
+- To add a formula column
+
+### User adds a formula column from an existing column
+
+### User deletes a formula column
+
+## Filtering
+
+In the `Filter` step, users apply filters to columns in the result table.
+
+### User adds a filter step
+
+- A user can filter selected columns at any point in the workflow by adding a `Filter` step.
+- The filter step will let users select any columns present in the result table until that point, including formula columns if available.
+- Once a column is selected, the user will have all the filtering conditions available for each data type. See [Filters Protoype](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?page-id=4612%3A39411&node-id=4612%3A39412&viewport=324%2C48%2C0.23&scaling=contain&starting-point-node-id=4612%3A39412&show-proto-sidebar=1).
+- A user might apply multiple filters to the same or other columns in the result table.
+- A user can add more than one filter step.
+
+[Wireframe 6](https://share.balsamiq.com/c/t8zJwPzsCvVqsDuAkQSXmk.png)
+
+## Summarization
+
+In the `Summarize` step, users set a summary column and perform aggregations on the result table.
+
+### User adds a summarization step
+
+- A user can summarize the result table at any point in the workflow by adding a `Summarization` step.
+- The summarization step will let users set a column as the summary column. By doing so, The system will aggregate the rest of the columns.
+- The summary column might or might not be a part of the result table—any formula columns created before the summarization step can be set as a summary column.
+- Once a summary column is set, the aggregated columns will be listed, each with a suggested aggregation function. Users can change the suggested functions.
+- A user can add more than one summarization step.
+
+[Wireframe 7](https://share.balsamiq.com/c/2e6TYLVzqf4pssmfpNqacx.png)
```

Comment:
This wireframe doesn't have anything related to summarization, I assume you'll update it.